src/pip/_internal/metadata/__init__.py

```diff
@@ -54,6 +55,31 @@ def _should_use_importlib_metadata() -> bool:
     return bool(getattr(importlib.metadata, "_PIP_USE_IMPORTLIB_METADATA", True))
 
 
+def _emit_pkg_resources_deprecation_if_needed() -> None:
+    if sys.version_info < (3, 11):
+        # All pip versions supporting Python<=3.11 will support pkg_resources,
+        # and pkg_resources is the default for these, so let's not bother users.
+        return
+
+    import importlib.metadata
+
+    if hasattr(importlib.metadata, "_PIP_USE_IMPORTLIB_METADATA"):
+        # The Python distributor has set the global constant, so we don't
+        # warn, since it is not a user decision.
+        return
```

Comment:
I'm not entirely sure we should not warn the user here.

Reply:
I'm not aware of any redistributor setting this attribute. Last time we reached out to our distro redistributors, none said they needed to use `pkg_resources` which implies that they didn't use this attribute either.